### PR TITLE
App

### PR DESCRIPTION
**Pull request by ivankzl**
Feature: React application that allows you to create, read, update and delete recipes and add/delete ingredients to i. It consumes the CRUD API from this repo: https://github.com/ivankzl/tk-recipes-backend

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import {
+  BrowserRouter,
+  Switch,
+  Route,
+  // @ts-ignore
```
Comment (giuband) [inline src/App.tsx]: You can easily solve these issues by adding `@types/react-router-dom` to the list of dev dependencies for your project

File: src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+} from "react-router-dom";
+
+import ScreensHome from './screens/Home';
+import AppHeader from './components/AppHeader/AppHeader';
+import ScreensRecipeList from './screens/Recipe/List';
+import ScreensRecipeCreate from './screens/Recipe/Create';
+import ScreensRecipeDetail from './screens/Recipe/Detail';
+import ScreensRecipeEdit from './screens/Recipe/Edit';
+import ScreensAbout from './screens/About';
+// @ts-ignore
```
Comment (giuband) [inline src/App.tsx]: Same here with `@types/styled-components`

File: src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import styled from 'styled-components';
+
+const MainContainer = styled.div`
+  margin-top:2rem;
```
Comment (giuband) [inline src/App.tsx]: Why mixing `rem` and `px`?

File: src/components/AppHeader/styled.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  width: 100%;
+  a {
+    float: left;
+    display: block;
+    color: #f2f2f2;
+    text-align: center;
+    padding: 14px 16px;
+    text-decoration: none;
+  }
+  a:hover {
```
Comment (giuband) [inline src/components/AppHeader/styled.tsx]: You usually want to apply the same styles to `:hover`, `:active` and `:focus`, to increase accessibility and make life easier to keyboard users 👌

File: src/components/Recipe/Form/Form.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  ingredients: string[];
+}
+
+function Form({
+  initialRecipe = { name: '', description: '', ingredients: [{ name: '' }] },
+  initialId = 0
+}: Props): ReactElement {
+  const { register, handleSubmit, formState: { errors } } = useForm<RecipeFormValues>();
+  const[name, updateName] = useInputState(initialRecipe?.name || '');
+  const[description, updateDescription] = useInputState(initialRecipe?.description || '');
```
Comment (giuband) [inline src/components/Recipe/Form/Form.tsx]: The spacing is a bit inconsistent, consider using prettier (you'll use it a lot here at TK)

File: src/components/Recipe/Form/useInputState.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { useState } from 'react';
+
+function useInputState(initialVal: any){
```
Comment (giuband) [inline src/components/Recipe/Form/useInputState.tsx]: Try to never use `any`, it kills the purpose of using typescript. In our apps you won't be allowed to use it.

File: src/components/Recipe/Item/Item.test.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  return { ...utils, history }
+}
+
+describe('<RecipeItem>', () => {
+  it('should render a recipe', async () => {
+    renderRecipeItem({ recipe });
+  })
+
+  it('should render a recipe name', async () => {
+    const { getByText } = renderRecipeItem({ recipe });
```
Comment (giuband) [inline src/components/Recipe/Item/Item.test.tsx]: Prefer using `screen.getByText` (which I saw you're already using in some other files)

File: src/components/RecipeSearch.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+const StyledLink = styled(Link)`
+    float: right !important;
+    padding: 6px !important;
+    margin-top: 8px !important;
+    margin-right: 16px !important;
+    font-size: 17px !important;
+`;
+
+class RecipeSearch extends Component {
```
Comment (giuband) [inline src/components/RecipeSearch.js]: - Why is this in `.js` and not in typescript? 
- avoid class components, you'll see that in our projects you'll be asked to always use functional components + hooks instead

File: src/screens/Recipe/Edit.test.tsx [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+        ingredients: [{ name: 'Flour' }],
+      });
+    }, 1000);
+  });
+
+  it('should update the recipe when submitted with empty name', async () => {
+    // @ts-ignore
+    getRecipe.mockResolvedValue({
+      id: 1,
+      name: 'Gnocci',
```
Comment (giuband) [inline src/screens/Recipe/Edit.test.tsx:73]: *Gnocchi 😄

File: src/App.tsx (hunk 1 above)
Comment (ivankzl) [reply src/App.tsx]: Nice, I've just removed the unnecessaries ts-ignores! Thanks :)

File: src/App.tsx (hunk 3 above)
Comment (ivankzl) [reply src/App.tsx]: You are right, fixed!

File: src/components/AppHeader/styled.tsx (hunk 4 above)
Comment (ivankzl) [reply src/components/AppHeader/styled.tsx]: I didn't know about that. Fixed. Thanks!

File: src/components/Recipe/Item/Item.test.tsx (hunk 7 above)
Comment (ivankzl) [reply src/components/Recipe/Item/Item.test.tsx]: Done, thanks

File: src/components/RecipeSearch.js (hunk 8 above)
Comment (ivankzl) [reply src/components/RecipeSearch.js]: I missed that one when refactoring everything to TS on the last stage. Thanks, updated

**Pull request merged**